### PR TITLE
use prop-types package

### PR DESCRIPTION
**Pull request by joeduncan**
replace deprecated `React.PropTypes` with `prop-types` package

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,7 +27,6 @@
     "build-es5": "mkdir -p dist && babel src --out-dir dist --plugins=transform-es2015-modules-commonjs --source-maps inline",
     "build-es6": "mkdir -p dist-es6 && babel src --out-dir dist-es6 --source-maps inline",
     "clean": "rm -fr dist/* dist-es6/*",
-    "clean": "rm -fr dist/* dist-es6/*",
```
Comment (ibgreen) [inline package.json]: Why remove the "clean" script? I'd prefer that we didn't. If there is a reason to do it, it should be done in a separate PR.

Comment (joeduncan) [inline package.json]: it was in there twice...but I see it was removed by last commit https://github.com/uber/react-map-gl/commit/6e3a934613a8711728c929cb85558b1ada073c6f

Comment (joeduncan) [inline package.json]: I'll rebase, give me a sec

**Comment by abmai** [comment]
Looks great, and looks like clean deletion is removed. Thanks, @joeduncan!
